ods_ci/tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot

```diff
@@ -40,13 +42,36 @@ Wait Until JupyterHub Spawner Is Ready
     Wait Until Page Contains Element    xpath:${JUPYTERHUB_DROPDOWN_XPATH}\[1]    timeout=15s
 
 Select Notebook Image
-    [Documentation]  Selects a notebook image based on a partial match of ${notebook_image} argument
-    [Arguments]    ${notebook_image}
-    #Wait Until Element Is Visible    xpath://div[@class="jsp-spawner__image-options"]
+    [Documentation]    Selects a notebook image based on a partial match of ${notebook_image} argument
+    ...                ${version} controls if the default or previous version is selected (default | previous)
+    [Arguments]    ${notebook_image}    ${version}=default
+    ${KFNBC_IMAGE_ROW} =    Set Variable    //input[contains(@id, "${notebook_image}")]
+    ${KFNBC_IMAGE_DROPDOWN} =    Set Variable    ${KFNBC_IMAGE_ROW}/../../div[contains(@class, "notebook-image-tags")]
     Wait Until Element Is Visible    xpath://div[.="Notebook image"]/..
-    Wait Until Element Is Visible    xpath://input[contains(@id, "${notebook_image}")]
-    Element Should Be Enabled    xpath://input[contains(@id, "${notebook_image}")]
-    Click Element    xpath://input[contains(@id, "${notebook_image}")]
+    Wait Until Element Is Visible    xpath=${KFNBC_IMAGE_ROW}
+    Element Should Be Enabled    xpath=${KFNBC_IMAGE_ROW}
+    IF    "${version}"=="default"
+        Click Element    xpath=${KFNBC_IMAGE_ROW}
+    ELSE IF    "${version}"=="previous"
+        Verify Version Dropdown Is Present    ${notebook_image}
+        Click Element    xpath=${KFNBC_IMAGE_ROW}/../..//button[.="Versions"]
+        Click Element    xpath=${KFNBC_IMAGE_DROPDOWN}//span[contains(text(), "Python v${PREVIOUS_PYTHON_VER}")]/../input
```

Comment:
this would work only if the image versions differs for python version. Are we sure it will always be the case?

Reply:
It is currently, but it's possible we'll need to update this in the future.
Right now the images don't have very descriptive names (just a string that looks like its tag) so I wanted to avoid using that, since I plan on opening a bug to get that changed